integrationTests/chainSimulator/bridge/executeAndBurn_test.go

```diff
@@ -0,0 +1,308 @@
+package bridge
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-go/config"
+	chainSim "github.com/multiversx/mx-chain-go/integrationTests/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/dtos"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	issuePrice = "5000000000000000000"
+)
+
+func TestChainSimulator_ExecuteWithMintAndBurnFungibleWithDeposit(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	token := "sov1-SOVTKN-1a2b3c"
+	tokenNonce := uint64(0)
+
+	mintTokens := make([]chainSim.ArgsDepositToken, 0)
+	mintTokens = append(mintTokens, chainSim.ArgsDepositToken{
+		Identifier: token,
+		Nonce:      tokenNonce,
+		Amount:     big.NewInt(123),
+		Type:       core.Fungible,
+	})
+	mintTokens = append(mintTokens, chainSim.ArgsDepositToken{
+		Identifier: token,
+		Nonce:      tokenNonce,
+		Amount:     big.NewInt(100),
+		Type:       core.Fungible,
+	})
+
+	depositTokens := make([]chainSim.ArgsDepositToken, 0)
+	depositTokens = append(depositTokens, chainSim.ArgsDepositToken{
+		Identifier: token,
+		Nonce:      tokenNonce,
+		Amount:     big.NewInt(12),
+		Type:       core.Fungible,
+	})
+	depositTokens = append(depositTokens, chainSim.ArgsDepositToken{
+		Identifier: token,
+		Nonce:      tokenNonce,
+		Amount:     big.NewInt(10),
+		Type:       core.Fungible,
+	})
+
+	simulateExecutionAndDeposit(t, mintTokens, depositTokens)
+}
+
+func TestChainSimulator_ExecuteWithMintAndBurnNftWithDeposit(t *testing.T) {
```

Comment:
Perhaps rename a bit more suggestive, since you "execute" nft + esdt, but only deposit nft